core/orchestrator.go

```diff
@@ -899,7 +912,7 @@ func (rtm *RemoteTranscoderManager) totalLoadAndCapacity() (int, int, int) {
 
 // Transcode does actual transcoding using remote transcoder from the pool
 func (rtm *RemoteTranscoderManager) Transcode(md *SegTranscodingMetadata) (*TranscodeData, error) {
-	currentTranscoder := rtm.selectTranscoder()
+	currentTranscoder := rtm.selectTranscoder(md.AuthToken.SessionId)
```

Comment:
WDYT about returning `errors.New("No transcoders available")` instead of returning `nil` from `selectTranscoder()` and handling the error case instead of the nil case?  Feels a little bit cleaner and we have the opportunity to do that now. 

Can also turn it into a constant since the error message doesn't take arguments. 

`var errNoTranscodersAvailable = errors.New("No transcoders available")`

Reply:
@kyriediculous, I have pushed the changes suggested here and above as well, could you please review how the error handling is now done here, is this okay? I'm not quite sure. Leaving everything unresolved so you can verify if this addresses everything and resolve them.

I am yet to look into the test related nil pointer error that occurs on the last test [here](https://github.com/livepeer/go-livepeer/pull/1849/files#diff-9734c79b7138aa52372c8c9714be7aba54b219fb23099883e1533f44d9f79385R350).